R/antigen.garnish_predict.R

```diff
@@ -105,13 +105,14 @@ parallel::mclapply(dt[, spc %>% unique], function(s){
     }
 
     blastdt <- blastdt[, nmer := nmer %>%
-                      stringr::str_replace_all(pattern = "-", replacement = "")] %>%
+                      stringr::str_replace_all(pattern = "-|\\*", replacement = "")] %>%
                .[, WT.peptide := WT.peptide %>%
-                      stringr::str_replace_all(pattern = "-", replacement = "")] %>%
+                      stringr::str_replace_all(pattern = "-|\\*", replacement = "")] %>%
                     .[nchar(nmer) == nchar(WT.peptide) & mismatch_length == 1] %>%
                       .[!is.na(nmer) & !is.na(WT.peptide)]
 
-    blastdt <- blastdt[!nmer %like% "B|U" & !WT.peptide %like% "B|U"]
+    # remove uncertain AA calls
+    blastdt <- blastdt[!nmer %like% "B|U|X|Z" & !WT.peptide %like% "B|U|X|Z"]
```

Comment:
instead of doing it this way, what do you think about white-listing known good AA symbols so we avoid all issues moving forward?

Reply:
Yes, I will add this.

Reply:
Done